lib/rubygems/commands/contents_command.rb

```diff
@@ -167,18 +167,16 @@ def show_files(files)
   end
 
   def spec_for(name)
-    spec = Gem::Specification.find_all_by_name(name, @version).last
-
-    return spec if spec
-
+    Gem::Specification.find_by_name(name, @version)
```

Comment:
`spec = Gem::Specification.find_all_by_name(name, @version).first` ?

Reply:
This method currently has bad logic (with `.last`) so I was trying to get away from it having _any_ of its own logic for choosing which of the gems to use, because that logic already exists elsewhere:
 * `Gem::Specification.find_all_by_name` is documented as `Returns every spec that matches +name+`, which then requires an additional logical step in this method to limit it to one spec.
 * `Gem::Specification.find_by_name` is documented as `Find the best specification matching a +name+` and already incorporates the necessary logic to find a single spec.

Does that thinking make sense?

But I'm certainly not very familiar with the RubyGems code so if you think simply switching from `.last` to `.first` is better, I can update.

Reply:
I prefer `.first` too for this fix, since it makes the fix more explicit and consistent with what we do in other commands, for example:

https://github.com/rubygems/rubygems/blob/98ca0d40f1f961a44c55695139ac1d7df7c087d8/lib/rubygems/commands/open_command.rb#L81

I'm happy to discuss improving/unifying this, but I think it should be a separate PR.

Reply:
Thank you, that other example is super helpful to see. I'm totally happy to stay consistent with that so have updated the PR accordingly.